llpc/tool/llpcPipelineBuilder.cpp

```diff
@@ -84,19 +84,26 @@ bool PipelineBuilder::isGraphicsPipeline() const {
 //
 // @returns : Result::Success on success, other status on failure.
 Result PipelineBuilder::build() {
-  BinaryData pipeline = {};
+  // Use vector of BinaryData for future pipeline type
+  SmallVector<BinaryData, 1> pipelines;
+  pipelines.push_back({});
   const bool isGraphics = isGraphicsPipeline();
   if (isGraphics) {
-    Result result = buildGraphicsPipeline(pipeline);
+    Result result = buildGraphicsPipeline(pipelines[0]);
     if (result != Result::Success)
       return result;
   } else {
-    Result result = buildComputePipeline(pipeline);
+    Result result = buildComputePipeline(pipelines[0]);
     if (result != Result::Success)
       return result;
   }
 
-  return decodePipelineBinary(&pipeline, &m_compileInfo, isGraphics);
+  for (const auto &pipeline : pipelines) {
+    auto result = decodePipelineBinary(&pipeline, &m_compileInfo, isGraphics);
```

Comment:
nit: could you change this to `Result result` to keep consistent with the surrounding code?